src/Effect/Ref.purs

```diff
@@ -13,7 +13,12 @@ import Effect (Effect)
 foreign import data Ref :: Type -> Type
 
 -- | Create a new mutable reference containing the specified value.
-foreign import new :: forall s. s -> Effect (Ref s)
+new :: forall s. s -> Effect (Ref s)
+new s = self (const s)
```

Comment:
I don't really like that `new` requires an unnecessary closure allocation with `const` on top of the ref allocation and then Effect thunk allocation.

Reply:
Yeah, I guess the whole thing is FFI already so one more function won't do any harm.